### DISCUSSION TITLE
Problem installing on QNAP

### DISCUSSION THREAD

**Discussion by lucaberti** [ente-io/ente#3143] [Q&A] [answered]
### Description

I'm trying to install a self hosted server instance on a QNAP NAS, but even following this instructions (https://github.com/ente-io/ente/blob/main/server/docs/docker.md) and using a old version of the compose.yaml file (since  "start_interval: 1s" is not supported by docker version 2.15 installend on the NAS) I cannot get things working. ente-minio-provision-1 exots almost immediatly with code 0.
Then I get this errors in musem:
```
ente-museum-1           | ERRO[0001]offer.go:46 NewOfferController Error reading config fileopen : no such file or directory
ente-museum-1           | ERRO[0001]offer.go:50 NewOfferController Error un-marshalling JSONunexpected end of JSON input
ente-museum-1           | ERRO[0001]billing.go:91 parsePricingFile Error reading file : open : no such file or directory
ente-museum-1           | ERRO[0001]billing.go:91 parsePricingFile Error reading file : open : no such file or directory
```

Finally when I try to connect to the web interface, I get this error:

```
WARN[0121]main.go:976 urlSanitizer Unknown API: /
```
Is there a way to ovecome this problems.

FYI NAS is a intel-Celeron-based one.

### What product are you using?

Ente Photos

### What platform are you using?

Web - Chrome, Server

**Comment by mnvr** [comment, marked as answer]
> I cannot get things working. ente-minio-provision-1 exots almost immediatly with code 0.

That is expected, it is not an error. That container is only run initially and exits immediately with success code 0 after its work is done.

> Then I get this errors in musem: 

These are also expected and not a problem. The `NewOfferController` and `parsePricingFile` are usually not need during self hosting.

> WARN[0121]main.go:976 urlSanitizer Unknown API: /

This is the real error. It means you are connecting to museum instead of the web app.

The web app is a different app that runs on a different endpoint (or a different port if you're running everything on localhost). When self hosting, there are at least three ports involved

* Museum (8080 by default)
* Web app (3000 by default)
* Minio (3200 by default)

When opening the web app, you need to connect to localhost:3000.

**Reply by lucaberti** [reply]
Thank you for the prompt answer. 
Still, I don't get any access to the web interface on port 3000.
Considering that port 8080 is already in use on my server, is there a way to move to another port?
And... in the compose.yaml file I see some standard password are used. I don't see any indication those have to be changed and when I try, I get a "sum check" error. 
Am I supposed to change those password? If yes, how? (I changed them both on compose.yaml and credentials.yaml)

Please note that I'm using the prebuild docker image...

**Comment by SSj-Saturn** [comment]
Same issue via Docker on Mac, I think the issue if we need the web app, it's not currently included in the Docker-Compose.yml?

**Comment by maxime1992** [comment]
Hi, has anyone found any answer? I'm running exactly into the same issue. Any help is much appreciated, thanks!

**Comment by andreas-bulling** [comment]
I also followed the instructions (https://ente.com/help/self-hosting/installation/compose) but I don't see any service listing on port 3000 after installation.

BTW - there is also no mentioning at all of port 3000 anywhere in the `museum.yaml` or `.env`
